### PR TITLE
Arc 1558 push app id to queue msg  sync utils

### PR DESCRIPTION
**Pull request by gxueatlassian**
**What's in this PR?**
Add gitHubAppConfig support for sync-util
This `sync-util` is used in backfill ( in api router and github configuration post ). 

**Why**
gitHubInstallationId is not unique for GHES

**Added feature flags**
N/A

**Affected issues**  
ARC-1558

**How has this been tested?**  
Unit test.

**Whats Next?**
N/A

### CODE REVIEW

File: src/sync/sync-utils.ts [hunk 1]
```diff
@@ -52,3 +59,42 @@ export const getCommitSinceDate = async (jiraHost: string, flagName: NumberFlags
...
+const ghesGitHubAppConfig = (app: GitHubServerApp): GitHubAppConfig => {
+	return {
+		gitHubAppId: app.id,
+		appId: app.appId,
+		uuid: app.uuid,
+		clientId: app.gitHubClientId,
+		gitHubBaseUrl: app.gitHubBaseUrl,
+		gitHubApiUrl: app.gitHubBaseUrl
+	};
+};
```
Comment (mboudreau) [inline src/sync/sync-utils.ts:100]: hm, I have a feeling that this config creation should be abstracted somewhere so anything can use it.

Comment (gxueatlassian) [reply src/sync/sync-utils.ts:100]: I was thinking exactly the same thing. But didn't do it in this PR, as currently there're three major places to have this config:

1. WebhookReceiverPost  --> no secrets, just id --> config pushed to context --> queue payload
2. GitHubServerMiddlware, --> secrets also available --> config saved in res.locals
3. uitl/get-github-client-config --> private keys available --> config used locally to createXXXClient.

And shapes are also a bit different as well.

Maybe let's discuss it and address all above in another PR?

Comment (mboudreau) [reply src/sync/sync-utils.ts:100]: sounds good :+1:

**Review by mboudreau — APPROVED** [review]
gj

File: src/sync/sync-utils.ts (hunk 1 above)
Comment (gxueatlassian) [reply src/sync/sync-utils.ts:100]: Ticket created. 
https://softwareteams.atlassian.net/browse/ARC-1582

**Pull request merged**